src/tools/schema2graph.ts

```diff
@@ -186,7 +197,7 @@ export class SchemaGraph {
       if (nestedSchema) {
         // We have a reference field. Generate a node for its nested schema and connect it into the
         // refs map to indicate that this node requires nestedNode's class to be generated first.
-        const nestedNode = this.createNodes(nestedSchema, particleSpec, source.child(field));
+        const nestedNode = this.createNodes(nestedSchema, particleSpec, source.child(field), fromVariable);
```

Comment:
Interesting! If I write
```
particle P
  foo: reads ~a with Person {
    name: Text,
    friends: [&Friend {
      name: Text,
      friendshipEstablished: Timestamp
    }]
  }
  bar: writes ~a
  baz: reads Friend {
    name: Text,
    friendshipEstablished: Timestamp
  }
```

Can `P_Baz` and `P_Foo_Friends` be backed by the same code generated entity? In other words is the `Friend` schema from the `foo` connection subject to the restrictions of the type variable-based schemas - i.e. you cannot instantiate from thin air?

I think so, because:

With `bar` connection you can write out modified versions of instances of ~a. When you make modifications to Person entities you've read from `foo`, you can modify the `friends` field. But the actual type of the `friends` field may be different in the fully resolved `~a`, e.g. the Friend may contain `age: Number` and `interests: [Text]` as well. Therefore when you make a copy of the entity of type ~a and you modify the friends field, you can only put references to the entities that came from the `foo` connection, and not for example from the `baz` connection or that you manufactured from thin air.

Therefore:
1) `P_Baz` is different than `P_Foo_Friends` - they cannot be backed by the same code generated entity.
2) You are propagating `fromVariable` correctly here.

-------

Someday we may look into havig type variables deeper in the schema e.g.:

```
particle P
  person: reads Person {
    name: Text,
    friends: [&~f with Friend {
      name: Text,
      friendshipEstablished: Timestamp
    }]
  }
  friend: writes ~f
```

And then we would need to look at type variables deeper in the entity, but I don't think it will work today and I think you can leave this code as is.

Reply:
> Can P_Baz and P_Foo_Friends be backed by the same code generated entity?

This is an excellent test case that I've missed; I'll incorporate it now.

Reply:
> 2. You are propagating fromVariable correctly here.

Thanks for your insight. I'm currently debating whether to associate `fromVariable` with the `SchemaNode` or the `SchemaSource`. Once I have tests nailed down, I'll explore each implementation option.

Reply:
> Someday we may look into havig type variables deeper in the schema 
That's interesting. This feature would offer a lot of expressive power. 

Hmm... is this case covered by [these tests](https://github.com/PolymerLabs/arcs/pull/5388/files#diff-dca0dfdc53d79b37f1d6ecec289c7fd1R692)? I'll think about it...